src/librustc_typeck/check/method/probe.rs

```diff
@@ -799,7 +807,25 @@ impl<'a, 'gcx, 'tcx> ProbeContext<'a, 'gcx, 'tcx> {
             return Err(MethodError::PrivateMatch(def, out_of_scope_traits));
         }
 
+        debug!("Probing for method names similar to {:?}",
+               self.method_name);
+
+        // Toggle the flag that allows for similar names to be found
+        // and search again
+        self.allow_similar_names = true;
+        self.inherent_candidates.clear();
+        self.extension_candidates.clear();
+        self.assemble_inherent_candidates();
+        self.assemble_extension_candidates_for_traits_in_scope(ast::DUMMY_NODE_ID)?;
+
+        let mut lev_candidates = Vec::with_capacity(self.inherent_candidates.len() +
+                                                    self.extension_candidates.len());
+        for lev_cand in self.inherent_candidates.iter().chain(self.extension_candidates.iter()) {
+            lev_candidates.push(lev_cand.item);
```

Comment:
You might want to try to pick each candidate, to avoid showing unusable candidates.

Reply:
I'm not sure how to do this best. To pick, them I'd have to extract all the close matching names, run a `probe` for each and call pick() ?